patches/api/0427-Add-Listing-API-for-Player.patch

```diff
@@ -0,0 +1,54 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Corey Shupe <coreyshupe101@gmail.com>
+Date: Wed, 11 Jan 2023 16:40:31 -0500
+Subject: [PATCH] Add Listing API for Player
+
+
+diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
+index de960716478477ce199526b8f860cfafa1541ee9..d3dcfdcc5a879ea5842306cdc811277900825171 100644
+--- a/src/main/java/org/bukkit/entity/Player.java
++++ b/src/main/java/org/bukkit/entity/Player.java
+@@ -1483,6 +1483,43 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
+     @org.jetbrains.annotations.ApiStatus.Experimental // Paper
+     public boolean canSee(@NotNull Entity entity);
+ 
++    // paper start
++    /**
++     * Determines if the {@code other} player is listed for {@code this}.
++     *
++     * @param other The other {@link Player} to check for listing.
++     * @return True if the {@code other} player is listed for {@code this}.
++     */
++    public boolean isListed(@NotNull Player other);
++
++    /**
++     * Sets the {@code other} player as listed or unlisted.
++     *
++     * @param plugin The {@link Plugin} which wants to mutate the {@code other} player's listing.
++     * @param other The other {@link Player} to set listed or unlisted.
++     * @param listed Whether the {@code other} player should be listed or de-listed.
++     */
++    public void setListed(@NotNull Plugin plugin, @NotNull Player other, boolean listed);
++
++    /**
++     * De-lists the {@code other} player.
++     *
++     * @param plugin The {@link Plugin} that is de-listing the {@code other} player.
++     * @param other The other {@link Player} to de-list.
++     * @return True if the {@code other} player was listed.
++     */
++    public boolean deListPlayer(@NotNull Plugin plugin, @NotNull Player other);
```

Comment:
not a big fan for this name... maybe can be a unique method with a bool param for add/remove?

Reply:
https://github.com/PaperMC/Paper/pull/8782/files#diff-7f2b831be83681583b338f6fb1dd559ffd5eec2e05ef3016ac7c66975a8e8b31R31